shadows/framework/src/main/java/org/robolectric/shadows/ShadowWifiManager.java

```diff
@@ -844,23 +845,22 @@ public void networksFoundFromPnoScan(List<ScanResult> scanResults) {
     }
   }
 
-  // Object needs to be used here since PnoScanResultsCallback is hidden. The looseSignatures spec
-  // requires that all args are of type Object.
   @Implementation(minSdk = TIRAMISU)
   @HiddenApi
   protected void setExternalPnoScanRequest(
-      Object ssids, Object frequencies, Object executor, Object callback) {
+      @ClassName("java.util.List<android.net.wifi.WifiSsid>") Object ssids,
```

Comment:
Does the `<android.net.wifi.WifiSsid>` actually work? Can this just be `List ssids`?

Reply:
The same pattern is used in other places, so I think it's fine to use that.
![Screenshot 2024-08-15 at 09 10 44](https://github.com/user-attachments/assets/15782088-2bc0-4602-bad1-8d6eda71f7a8)

Reply:
actually I think the annotation processor do ignore the type parameter when parsing `ClassName` annotation, so when the real type is is say `List<SomeAndroidType>`, then using `@ClassName("java.util.List") Object`, or `@ClassName("java.util.List<SomeAndroidType>") Object` or `@ClassName("java.util.List<AnyStringHere>") Object` all works.

Reply:
Does it work if you do:

```java
  @Implementation(minSdk = TIRAMISU)
  @HiddenApi
  protected void setExternalPnoScanRequest(
      List<?> ssids,
      int[] frequencies,
      Executor executor,
      @ClassName("android.net.wifi.WifiManager$PnoScanResultsCallback") Object callback) {
```

Reply:
Using `List` instead of `@ClassName` seem to work fine, yes

Reply:
yeah, this is really weird, only if I removed type argument, then the test will pass

Reply:
I didn't look too deep in the code, but the implementation of the `@ClassName` management is different in 'ShadowWrangler` in `master` and in #8829.
So maybe there are some changes to do there?

Reply:
juts tried, not able to retrieve the type argument from the `Class<?>` representation of the parameter (of the shadowed method), looks like the type argument info is gone in runtime

Reply:
> https://github.com/robolectric/robolectric/pull/8829/files#diff-a2f9697503e542e27ab7f01655790bc6ea3f56dfa56c26c5f1af5db3e5a3a27a Looks like I can use `@ClassName("java.util.List<android.net.wifi.WifiSsid>")` in my previous PR.

I just doubled compared your `ShadowWrangler` implementation with my merged PR, emm, imo your implementation did not take type argument (of the parameter of shadowed method) into consideration either, but why your tests passed in those `ShadowWifiManagerTest.setExternalPnoScanRequest(...)`

Reply:
wow, I read your prototype, looks like it ignored the parameter Class<?>(s)  retrieved from `java.lang.invoke.MethodType`, instead you search the shadowed class for the matched  `java.lang.reflect.Method` and in this returned `Method`, its parameter will have type argument in runtime; thus with this info, you successfully compare with the value in `ClassName` annotation. Will this strategy work with say nested generic? say `List<List<String>>`

Reply:
wow, nice idea.
<img width="1106" alt="Capture" src="https://github.com/user-attachments/assets/7f4f9cb0-f1af-4f17-b334-36a01fe055a1">